pkg/api/file.go

```diff
@@ -201,6 +202,61 @@ func (s *server) fileUploadHandler(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+// fileUploadInfo contains the data for a file to be uploaded
+type fileUploadInfo struct {
+	name        string // file name
+	size        int64  // file size
+	contentType string
+	reader      io.Reader
+}
+
+// storeFile uploads the given file and returns its reference
+// this function was extracted from `fileUploadHandler` and should eventually replace its current code
+func storeFile(ctx context.Context, fileInfo *fileUploadInfo, s storage.Storer) (swarm.Address, error) {
+	v := ctx.Value(toEncryptContextKey{})
```

Comment:
ideally this would not even be here, and the `toEncryptContextKey` would be exposed in the `sctx` library, and then you would check for the value on the context inside the `SplitWriteAll` function in `file` package. I guess we can live without it for now though since this is a bit of a wider change.

Reply:
is this happening in the rest of the codebase then? do we have an issue to tackle this?

Reply:
no... let's leave it for now